.github/workflows/ci.yml

```diff
@@ -1,29 +1,114 @@
-on: [push, pull_request]
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+  release:
+    types: [published]
 
 jobs:
-  build:
+  build-and-test:
```

Comment:
We want to run the old code every time we push, and everything related to executables only in these situations. I think that we can accomplish this by separating the jobs. But, before merging, let's also make sure that this can be executed successfully, so run the build on push too, then revert that before merging.